src/Task.cc

```diff
@@ -3701,6 +3701,57 @@ static void copy_mem_mapping(Task* from, Task* to, const KernelMapping& km) {
   }
 }
 
+static void copy_mem_mapping_just_used(Task* from, Task* to, const KernelMapping& km)
+{
+  ScopedFd& fd = from->vm()->pagemap_fd();
+  if (!fd.is_open()) {
```

Comment:
Fall back to `copy_mem_mapping()` in this case.

Also, call `other->open_pagemap_fd()` here, which should also return the `ScopedFd`.